spec/app/domain/authentication/validate_security_spec.rb

```diff
@@ -251,4 +256,57 @@ def resource_class(returned_resource)
       expect { subject }.to raise_error(Errors::Authentication::Security::AccountNotDefined)
     end
   end
+
+  context "with the default role class" do
+    let(:admin_roleid) { [test_account, 'user', 'admin'].join(':') }
+    let(:admin_role_double) { double('admin_role_double') }
+
+    let(:ws_name) { 'service1' }
+    let(:ws_id) { ['conjur', fake_authenticator_name, ws_name].join('/') }
+    let(:ws_resourceid) { [test_account, 'webservice', ws_id].join(':') }
+    let(:ws_double) { double('ws_resource_double') }
+    
+    let(:user_id) { 'some-user' }
+    let(:user_roleid) { [test_account, 'user', user_id].join(':') }
+    let(:user_role_double) { double('user_role_double') }
+
+    before do 
+      allow(Role).to receive(:[]).with(admin_roleid).and_return(admin_role_double)
+      allow(Role).to receive(:roleid_from_username).and_return(user_roleid)
+      
+      allow(Resource).to receive(:[]).with(ws_resourceid).and_return(ws_double)
+
+      allow(user_role_double).to receive(:allowed_to?).with('authenticate', ws_double).and_return(true)
+    end
+    
+    it "role lookups are not cached across validations" do
+
+      # Simulate two validations. For the first, the role should not
+      # be found.
+      allow(Role).to receive(:[]).with(user_roleid).and_return(nil)
+      expect { subject.(
+                   webservice: webservice(ws_name),
+                   account: test_account,
+                   user_id: user_id,
+                   enabled_authenticators: two_authenticator_env
+                 )
+      }.to raise_error(Errors::Authentication::Security::UserNotDefinedInConjur)
```

Comment:
I'm confusing how this is working because I couldn't find the definition for this `subject`.  The definitions above it in the file seem to be for subjects defined in other scopes, which are always out of scope by the time we get here.  

Related: We should use named subjects for clarity.

Reply:
When you `Rspec.describe` a Class, as this test does, the subject of the test is an instance of the Class. It's not clear to me that it's necessary to repeat this.

Reply:
Ah right, forgot about that.  Ok, I hereby *amend* the above comment to:

We should treat this consistently.  Ie, in the test cases above this one we're explicitly creating the subject again with the mocks.  I believe if once you make the change below (discussed on slack) this will happen automatically.